test/getTestFrameworkName.spec.ts

```diff
@@ -56,12 +66,27 @@ describe("getTestFrameworkName()", () => {
     assert.equal(getTestFrameworkName(packageJson), "mocha");
   });
 
-  it("Should return node:test", () => {
-    const packageJson = {
-      scripts: {
-        test: "node --test"
-      }
-    };
-    assert.equal(getTestFrameworkName(packageJson), "node:test");
+  describe("node:test", () => {
+    it("Should return node:test for test script", () => {
+      const packageJson = {
+        name: "test",
+        version: "1.0.0",
+        scripts: {
+          test: "node --test"
+        }
+      };
+      assert.equal(getTestFrameworkName(packageJson), "node:test");
+    });
+
+    it("Should return node:test for test-only script", () => {
+      const packageJson = {
+        name: "test",
+        version: "1.0.0",
+        scripts: {
+          "test-only": "glob -c \"node --loader=esmock --no-warnings --test-concurrency 1 --test\" \"test/**/*.test.js\""
+        }
+      };
+      assert.equal(getTestFrameworkName(packageJson), "node:test");
+    });
```

Comment:
Can you add tests for TypeScript ? (with `tsx`)

Reply:
Like this? ```it("Should return node:test if tsx is in dependencies", () => {
    const packageJson = {
      name: "test",
      version: "1.0.0",
      dependencies: {
        tsx: "^3.0.0"
      }
    };
    assert.equal(getTestFrameworkName(packageJson), "node:test");
  });```

Reply:
No we don't care the dependency, like this:
```ts
it("Should return node:test for test script when using tsx", () => {
      const packageJson = {
        name: "test",
        version: "1.0.0",
        scripts: {
          test: "tsx --test"
        }
      };
      assert.equal(getTestFrameworkName(packageJson), "node:test");
    });
```

Reply:
ho ok